### PR TITLE
Merge `24.4.x` back into `main`

### PR DESCRIPTION
**Pull request by jezdez**
Xref #13844

> [!WARNING]
> DO NOT SQUASH MERGE!

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -24,6 +37,7 @@
 * Fix excess resource usage by log handling when fetching repodata. (#13541 via #13628)
 * Re-enable `--subdir` and `--platform` flags to be available for `conda env create` command. (#13632)
 * Fix `__archspec` virtual package on Windows to return microarchitecture instead of the default `x86_64`. (#13641)
+* Check `Content-Length` is nonzero before calculating progress, avoiding a possible `ZeroDivisionError`. (#13653, #13671)
```
Comment (beeankha) [inline CHANGELOG.md:40]: Is this in here accidentally? Or did it get missed for the previous release and now it's getting added into the changelog retroactively?

Comment (dholth) [reply CHANGELOG.md:40]: It got missed in the previous notes.

Comment (jezdez) [reply CHANGELOG.md:40]: See https://github.com/conda/conda/pull/13871/commits/7921682d70a3e685d9d427e79c882e1b902b3252

Comment (jezdez) [reply CHANGELOG.md:40]: I started the 24.4.x branch from 24.3.x, that's why it's in here.

**Review by kenodegard — APPROVED** [review]

**Review by beeankha — APPROVED** [review]

**Pull request merged**